test/system/session/test_special_values.py

```diff
@@ -0,0 +1,877 @@
+
+
+@parametrize(
+    [
+        # All columns have 30 rows, all non-special values are equal to 1.
+        Case("int_noop")(
+            # Column "int_no_null" has only regular values.
+            affected_columns=["int_no_null"],
+            measure_column="int_no_null",
+            low=0,
+            high=1,
+            expected_sum=30,
+        ),
+        Case("int_drop_nulls")(
+            # Column "int_nulls" has one null value and 29 1s.
+            affected_columns=["int_nulls"],
+            measure_column="int_nulls",
+            low=0,
+            high=100,
+            expected_sum=29,
+        ),
+        Case("float_drop_nulls")(
+            # Column "float_nulls" has one null value and 29 1s.
+            affected_columns=["float_nulls"],
+            measure_column="float_nulls",
+            low=0,
+            high=100,
+            expected_sum=29,
+        ),
+        Case("float_drop_nan")(
+            # Column "float_nans" has one nan value and 29 1s.
+            affected_columns=["float_nans"],
+            measure_column="float_nans",
+            low=0,
+            high=100,
+            expected_sum=29,
+        ),
+        Case("float_drop_both")(
+            # Column "float_all_special" has 26 1s, one null, one nan, one negative
+            # infinity (clamped to 0), one positive infinity (clamped to 100).
+            affected_columns=["float_all_special"],
+            measure_column="float_all_special",
+            low=0,
+            high=100,
+            expected_sum=126,
+        ),
+        Case("drop_other_columns")(
+            # Column "float_infs" has 26 1s, two negative infinities (clamped to 0) and
+            # two positive infinities (clamped to 100). But dropping rows from columns
+            # "string_nulls", "float_nulls", "float_nans", "date_nulls" and "time_nulls"
+            # should remove five rows, leaving just 21 normal values.
+            affected_columns=[
+                "string_nulls",
+                "float_nulls",
+                "float_nans",
+                "date_nulls",
+                "time_nulls",
+            ],
+            measure_column="float_infs",
+            low=0,
+            high=100,
+            expected_sum=221,
+        ),
+        Case("drop_all_with_none")(
+            # When called with no argument, replace_null_and_nan should drop all rows
+            # that have null/nan values anywhere, which leaves 24 1s even if we're
+            # summing a column without nulls.
+            affected_columns=None,
+            measure_column="int_no_null",
+            low=0,
+            high=1,
+            expected_sum=24,
+        ),
+        Case("drop_all_with_empty_list")(
+            # Same thing with an empty list.
+            affected_columns=[],
+            measure_column="float_nulls",
+            low=0,
+            high=1,
+            expected_sum=24.0,
+        ),
+    ]
```

Comment:
This could probably use a case to test auto-dropping - none of these cases trigger the rewrite rule at the moment.